tests/functional/adapter/test_constraints.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+
+from dbt.tests.adapter.constraints.test_constraints import (
+    BaseTableConstraintsColumnsEqual,
+    BaseViewConstraintsColumnsEqual,
+    BaseIncrementalConstraintsColumnsEqual,
+    BaseConstraintsRuntimeDdlEnforcement,
+    BaseConstraintsRollback,
+    BaseIncrementalConstraintsRuntimeDdlEnforcement,
+    BaseIncrementalConstraintsRollback,
+    BaseModelConstraintsRuntimeEnforcement,
+)
+
+
+class DuckDBColumnEqualSetup:
+    @pytest.fixture
+    def int_type(self):
+        return "INT"
+
+    @pytest.fixture
+    def string_type(self):
+        return "VARCHAR"
+
+    @pytest.fixture
+    def data_types(self, schema_int_type, int_type, string_type):
+        # sql_column_value, schema_data_type, error_data_type
+        return [
+            ["1", schema_int_type, int_type],
+            ["'1'", string_type, string_type],
+            ["true", "bool", "BOOL"],
+            ["'2013-11-03 00:00:00-07'::timestamp", "TIMESTAMP", "TIMESTAMP"],
```

Comment:
@jtcohen6 and @sungchun12: noting that I could not for the life of me figure out how to configure this so that it would work with the `timestamptz` typing; the resulting data type in DuckDB will be `TIMESTAMP WITH TIMEZONE` and I'm not sure if the spaces are messing things up, or something like that?

If either of you have a moment to take a look and figure out how it's supposed to work, that would be great, but if not nbd-- I think we can live without it for 1.5.0.

Reply:
gotcha-- thank you, will give it a shot locally now!